test/test_test_app.rb

```diff
@@ -0,0 +1,50 @@
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+#
+
+require('minitest/autorun')
+
+require_relative('../test_app')
+
+def app_manifest_path(project_root, podspec_path)
+  File.join(project_root, podspec_path, 'ReactTestApp-Resources.podspec.json')
+end
+
+class TestTestApp < Minitest::Test
+  def test_resources_pod_returns_spec_path
+    assert_nil(resources_pod('/'))
+    assert_nil(resources_pod('.'))
+
+    project_root = File.join(__dir__, 'fixtures', 'without_resources')
+    assert_nil(resources_pod(project_root))
+
+    project_root = File.join(__dir__, 'fixtures', 'without_resources', 'ios')
+    assert_nil(resources_pod(project_root))
+
+    project_root = File.join(__dir__, 'fixtures', 'with_resources')
+    assert_equal('.', resources_pod(project_root))
+
+    project_root = File.join(__dir__, 'fixtures', 'with_resources', 'ios')
+    assert_equal('..', resources_pod(project_root))
+  end
+
+  def test_resources_pod_writes_podspec
+    [
+      File.join(__dir__, 'fixtures', 'with_resources'),
+      File.join(__dir__, 'fixtures', 'with_resources', 'ios')
+    ].each do |project_root|
+      podspec_path = resources_pod(project_root)
+      manifest_path = app_manifest_path(project_root, podspec_path)
+      manifest = JSON.parse(File.read(manifest_path))
+
+      assert_equal(2, manifest['resources'].length)
+      assert_equal('dist/assets', manifest['resources'][0])
+      assert_equal('dist/main.jsbundle', manifest['resources'][1])
+
+      File.delete(manifest_path)
```

Comment:
iirc these assertions will raise an exception and when that happens this line is never reached. Doing this from a ‘finally’ clause will ensure it always happens.

```suggestion
  ensure
      File.delete(manifest_path)
```

Reply:
Is this a new feature? RuboCop is yelling at me:
```
test/test_test_app.rb:47:3: E: Lint/Syntax: unexpected token kENSURE
(Using Ruby 2.3 parser; configure using TargetRubyVersion parameter, under AllCops)
  ensure
  ^^^^^^
```

Reply:
Hmm, perhaps the ability to have each `do … end` block be implicit ‘try’ blocks is, yeah. Then do:

```ruby
begin
  # ...
ensure
  File.delete(manifest_path)
end
```

Reply:
Thanks, that was it. The implicit behaviour was apparently introduced in 2.5.

Reply:
Gotcha. It was always there in the scope of a method, which made the experience inconsistent, so glad they added it 👍